java/ingest/ingest-runner/src/test/java/sleeper/ingest/impl/IngestCoordinatorUsingDirectWriteBackedByArrowIT.java

```diff
@@ -15,131 +15,244 @@
+        DynamoDBStateStore stateStore = new DynamoDBStateStoreCreator(
+                TABLE_NAME, recordListAndSchema.sleeperSchema, dynamoDB).create();
         PartitionTree tree = new PartitionsBuilder(recordListAndSchema.sleeperSchema)
                 .rootFirst("root")
                 .splitToNewChildren("root", "left", "right", 0L)
                 .buildTree();
+        stateStore.initialise(tree.getAllPartitions());
+        String ingestLocalWorkingDirectory = createTempDirectory(temporaryFolder, null).toString();
+        Instant stateStoreUpdateTime = Instant.parse("2023-08-08T11:20:00Z");
+        IngestCoordinatorTestParameters parameters = createTestParameterBuilder()
+                .fileNames(List.of("leftFile", "rightFile"))
+                .fileUpdatedTimes(() -> stateStoreUpdateTime)
+                .stateStore(stateStore)
+                .schema(recordListAndSchema.sleeperSchema)
+                .workingDir(ingestLocalWorkingDirectory)
+                .build();
 
-        test(recordListAndSchema, tree, arrow -> arrow
+        // When
+        Consumer<ArrowRecordBatchFactory.Builder<Record>> arrowConfig = config -> config
                 .workingBufferAllocatorBytes(16 * 1024 * 1024L)
                 .batchBufferAllocatorBytes(4 * 1024 * 1024L)
-                .maxNoOfBytesToWriteLocally(128 * 1024 * 1024L)
-        ).ingestAndVerify(keyToPartitionNoMappingFn, partitionNoToExpectedNoOfFilesMap);
+                .maxNoOfBytesToWriteLocally(128 * 1024 * 1024L);
+        TestIngestType ingestType = directWriteBackedByArrowWriteToLocalFile(arrowConfig);
+        ingestRecords(recordListAndSchema, parameters, ingestType);
+
+        // Then
+        List<FileInfo> actualFiles = stateStore.getActiveFiles();
+        FileInfoFactory fileInfoFactory = FileInfoFactory.builder()
+                .partitionTree(tree)
+                .lastStateStoreUpdate(stateStoreUpdateTime)
+                .schema(recordListAndSchema.sleeperSchema)
+                .build();
+        FileInfo leftFile = fileInfoFactory.leafFile(ingestType.getFilePrefix(parameters) +
+                "/partition_left/leftFile.parquet", 10000, -10000L, -1L);
+        FileInfo rightFile = fileInfoFactory.leafFile(ingestType.getFilePrefix(parameters) +
+                "/partition_right/rightFile.parquet", 10000, 0L, 9999L);
+
+        List<Record> actualRecords = readMergedRecordsFromPartitionDataFiles(
+                recordListAndSchema.sleeperSchema, actualFiles, configuration);
+
+        assertThat(actualFiles).containsExactlyInAnyOrder(leftFile, rightFile);
+        assertThat(actualRecords).containsExactlyInAnyOrderElementsOf(recordListAndSchema.recordList);
+        assertThat(actualRecords).extracting(record -> record.getValues(List.of("key0")))
+                .containsExactlyInAnyOrderElementsOf(LongStream.range(-10000, 10000).boxed()
+                        .map(List::<Object>of)
+                        .collect(Collectors.toList()));
```

Comment:
The separate assertion on the key values is usually used to assert the order of the records, and how they are separated between files. Please can we do that for each of the tests?

Reply:
It looks like we can assert the files' records separately for this test?